### PR TITLE
CORE-473: hotfix/CORE-473-add-api-helm-deploy

### PR DESCRIPTION
**Pull request by chrisnankervis**
**What Was Implemented**
* Adds sidekiq chart to package.sh
* Allows service annotations to be configured dynamically.

**Related Pull Requests**
* https://github.com/ActivePipe/api.activepipe.com/pull/2771
* https://github.com/ActivePipe/k8s-aws/pull/26
* https://github.com/ActivePipe/docker-k8s/pull/1

**Related JIRA Tickets**
* https://thedigitalgroup.atlassian.net/browse/SO-1147

### CODE REVIEW

File: api/templates/ingress.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
 {{- end }}
   rules:
   {{- range .Values.ingress.hosts }}
-    - host: {{ . }}
+    - host: {{ . | quote }}
```
Comment (Allen-Ning) [inline api/templates/ingress.yaml]: not sure what is this doing?

Comment (chrisnankervis) [reply api/templates/ingress.yaml]: This fixed an issue that I was running into with wildcard hostnames, see the issue below for more information ✨ 

https://github.com/helm/helm/issues/3936

File: api/templates/deployment.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
   namespace: {{ .Values.app.environment }}
-  name: {{ template "api.name" . }}
+  name: "{{ template "api.name" . }}-{{ .Values.app.environment }}"
```
Comment (Allen-Ning) [inline api/templates/deployment.yaml]: i give a try on my minikube. It seems like we don't need the quotes

File: sidekiq/templates/deployment.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
 apiVersion: apps/v1beta2
 kind: Deployment
 metadata:
-  name: {{ template "sidekiq.fullname" . }}
+  namespace: {{ .Values.app.environment }}
+  name: "{{ template "sidekiq.name" . }}-{{ .Values.app.environment }}"
```
Comment (Allen-Ning) [inline sidekiq/templates/deployment.yaml]: same here for quotes

**Pull request merged**